### PR TITLE
feat: adds orgs and starred repos to schema

### LINKED ISSUE #899: [Showcases API] Add Organizations and Starred Repositories To The Owner/Github User Schema
# Acceptance Criteria

Profile information

For the profile page, we wanted to fairly closely resemble the GitHub user page. This section is focused on the user “card”, where you can see profile information about the user. In order to get the stars and organizations for a certain user, we had to make 2 extra API calls which were:

- [List organizations for a user](https://docs.github.com/en/rest/orgs/orgs#list-organizations-for-a-user)
- [List repositories starred by a user](https://docs.github.com/en/rest/activity/starring#list-repositories-starred-by-a-user)

### PR DESCRIPTION
**Pull request by WillHutt**
Resolves #899 

- [x] Adds orgs
- [x] Adds stars

### Screenshot

image

### CODE REVIEW

**Review by danecando — APPROVED** [review]
Wondering if the graphql.ts file is generated and if so do we want to check it in to source control? Everything LGTM to me though

File: angular-apollo-tailwind/src/main.ts [hunk 1]
```diff
@@ -11,4 +11,3 @@ if (environment.production) {
 platformBrowserDynamic()
   .bootstrapModule(AppModule)
   .catch((err) => console.error(err));
```
Comment (Amdrel) [inline angular-apollo-tailwind/src/main.ts:13]: Nitpick: Removing unrelated and likely unintended whitespace change.

```suggestion
platformBrowserDynamic()
  .bootstrapModule(AppModule)
  .catch((err) => console.error(err));

```

File: starter-dev-backend/src/datasources/github-api.ts [hunk 2]
```diff
@@ -3,6 +3,9 @@ import {
   WillSendRequestOptions,
 } from '@apollo/datasource-rest';
 
+// TODO: follow up PR will cover formatters and testing
+// import { repoFormatter, orgFormatter, ownerFormatter } from 'src/formatters';
```
Comment (Amdrel) [inline starter-dev-backend/src/datasources/github-api.ts:7]: Blocker: Let's add a ticket number to the ticket being referenced here.

File: starter-dev-backend/src/datasources/github-api.ts [hunk 3]
```diff
@@ -19,20 +22,45 @@ export class GitHubAPI extends RESTDataSource {
...
+  }
+
+  async getOrgRepos(login: string) {
+    return await this.get(`/orgs/${login}`);
+  }
+
+  // https://docs.github.com/en/rest/orgs/orgs?apiVersion=2022-11-28#list-organizations-for-a-user
+  async getOrgs(login: string) {
+    const data = await this.get(`/users/${login}/orgs`);
+    // return orgFormatter(data);
```
Comment (Amdrel) [inline starter-dev-backend/src/datasources/github-api.ts:36]: Suggestion: Either remove unused and commented out code or reference a ticket number if related to the aforementioned formatting and testing ticket.

File: starter-dev-backend/src/datasources/github-api.ts [hunk 4]
```diff
@@ -19,20 +22,45 @@ export class GitHubAPI extends RESTDataSource {
...
+
+  async getOwnerStarCount(login: string) {
+    return await this.get(`/users/${login}/starred`);
+  }
+
   // https://docs.github.com/en/rest/users/users?apiVersion=2022-11-28#about-the-users-api
   async getOwner() {
-    return await this.get('/user');
+    const data = await this.get('/user');
+    // return ownerFormatter(data);
```
Comment (Amdrel) [inline starter-dev-backend/src/datasources/github-api.ts:47]: Suggestion: Either remove unused and commented out code or reference a ticket number if related to the aforementioned formatting and testing ticket.

File: starter-dev-backend/src/datasources/github-api.ts [hunk 5]
```diff
@@ -19,20 +22,45 @@ export class GitHubAPI extends RESTDataSource {
...
+    return data;
   }
 
   // https://docs.github.com/en/rest/repos/repos?apiVersion=2022-11-28#get-a-repository
   async getRepo(owner: string, repoName: string) {
-    return await this.get(`/repos/${owner}/${repoName}`);
+    const data = await this.get(`/repos/${owner}/${repoName}`);
+    // return repoFormatter(data);
```
Comment (Amdrel) [inline starter-dev-backend/src/datasources/github-api.ts:54]: Suggestion: Either remove unused and commented out code or reference a ticket number if related to the aforementioned formatting and testing ticket.

File: starter-dev-backend/src/datasources/github-api.ts [hunk 6]
```diff
@@ -19,20 +22,45 @@ export class GitHubAPI extends RESTDataSource {
...
+    return data;
   }
 
   // https://docs.github.com/en/rest/repos/repos?apiVersion=2022-11-28#list-repositories-for-a-user
   async getRepos(username: string, perPage?: string) {
-    return await this
+    const data = await this
       .get(`/users/${username}/repos?sort=updated&per_page=${perPage}
     `);
+    // return repoFormatter(data);
```
Comment (Amdrel) [inline starter-dev-backend/src/datasources/github-api.ts:63]: Suggestion: Either remove unused and commented out code or reference a ticket number if related to the aforementioned formatting and testing ticket.

File: starter-dev-backend/src/schema/github/github.typedefs.ts [hunk 7]
```diff
@@ -21,14 +21,14 @@ export const githubTypeDefs = gql`
   """
   A Owner object used in Repo
   """
```
Comment (Amdrel) [inline starter-dev-backend/src/schema/github/github.typedefs.ts:23]: Suggestion: Update comment to reflect schema change.

```suggestion
  """
  A User object used in Repo
  """
```

File: starter-dev-backend/src/schema/github/github.typedefs.ts [hunk 8]
```diff
@@ -21,14 +21,14 @@ export const githubTypeDefs = gql`
...
-  type Owner {
+  type User {
     login: String
   }
 
   """
   A User object
   """
```
Comment (Amdrel) [inline starter-dev-backend/src/schema/github/github.typedefs.ts:30]: Suggestion: Update comment to reflect schema change.

```suggestion
  """
  An Owner object
  """
```

File: starter-dev-backend/src/schema/github/github.typedefs.ts [hunk 9]
```diff
@@ -37,15 +37,28 @@ export const githubTypeDefs = gql`
     location: String
     login: String
     name: String
+    orgs: [Orgs]
+    starred_url: String
     twitter_username: String
   }
 
+  """
+  A Organization object used in Owner
```
Comment (Amdrel) [inline starter-dev-backend/src/schema/github/github.typedefs.ts:46]: Suggestion: Grammar 😛

```suggestion
  An Organization object used in Owner
```

File: starter-dev-backend/types/graphql.ts [hunk 10]
```diff
@@ -59,25 +79,17 @@ export type Repo = {
...
-  owner?: Maybe<Owner>;
+  owner?: Maybe<User>;
   private?: Maybe<Scalars['Boolean']>;
   stargazers_count?: Maybe<Scalars['Int']>;
   title?: Maybe<Scalars['String']>;
   updated_at?: Maybe<Scalars['String']>;
 };
 
-/** A User object */
+/** A Owner object used in Repo */
```
Comment (Amdrel) [inline starter-dev-backend/types/graphql.ts:89]: Suggestion: Grammar 😛

```suggestion
/** An Owner object used in Repo */
```

**Review by Amdrel — CHANGES_REQUESTED** [review]
Although there's no tests in this PR and it is required by the guidelines, it does not appear that this subproject has any testing infrastructure set up yet. I think it would be outside of the scope of this PR to set that all up and that can be handled in a future ticket. Other than that I see no glaring issues of note.

The comments that were left are mainly nitpicks that should be easily resolvable.

**Review by morgnism — APPROVED** [review]
LGTM

**Review by Amdrel — APPROVED** [review]

**Pull request merged**